From 33a5e52fa9dec25bc57a99883d6a675f36cc0543 Mon Sep 17 00:00:00 2001
From: Tri Vo <trong@google.com>
Date: Thu, 20 Jun 2019 18:04:53 -0700
Subject: ANDROID: gki_defconfig: workaround to enable configs

We're interested in following configs, which can't be selected by
themselves:
CONFIG_GPIOLIB_IRQCHIP
CONFIG_GENERIC_PINCONF
CONFIG_PINCONF
CONFIG_PINMUX
CONFIG_GENERIC_TRACER
CONFIG_MFD_CORE

As a workaround, enable configs that select the above:
CONFIG_PINCTRL_AMD
CONFIG_MFD_ACT8945A
CONFIG_FUNCTION_TRACER

Bug: 135666008
Bug: 135247530
Change-Id: If348f54fc11ab2c35ed2241d50327d8ce3b0b72e
Signed-off-by: Tri Vo <trong@google.com>
---
 abi_gki_aarch64.xml              | 221872 ++++++++++++++--------------
 arch/arm64/configs/gki_defconfig |      4 +-
 2 files changed, 110333 insertions(+), 111543 deletions(-)

Index: common/arch/arm64/configs/gki_defconfig
===================================================================
--- common.orig/arch/arm64/configs/gki_defconfig
+++ common/arch/arm64/configs/gki_defconfig
@@ -276,12 +276,13 @@ CONFIG_HW_RANDOM_VIRTIO=y
 # CONFIG_I2C_HELPER_AUTO is not set
 CONFIG_SPMI=y
 CONFIG_PINCTRL=y
-CONFIG_GPIOLIB=y
+CONFIG_PINCTRL_AMD=y
 # CONFIG_HWMON is not set
 CONFIG_THERMAL=y
 CONFIG_THERMAL_GOV_USER_SPACE=y
 CONFIG_CPU_THERMAL=y
 CONFIG_DEVFREQ_THERMAL=y
+CONFIG_MFD_ACT8945A=y
 CONFIG_MFD_SYSCON=y
 CONFIG_REGULATOR=y
 CONFIG_MEDIA_SUPPORT=y
@@ -400,6 +401,7 @@ CONFIG_SOFTLOCKUP_DETECTOR=y
 # CONFIG_DETECT_HUNG_TASK is not set
 CONFIG_PANIC_TIMEOUT=5
 CONFIG_SCHEDSTATS=y
+CONFIG_FUNCTION_TRACER=y
 # CONFIG_RUNTIME_TESTING_MENU is not set
 CONFIG_CORESIGHT=y
 CONFIG_CORESIGHT_STM=y
